### PR TITLE
[MDX] Support remark-rehype options from Astro Markdown config

### PR DESCRIPTION
**Pull request by backflip**
## Changes

- Uses remark-rehype options from astro.config.mjs
- Based on https://github.com/withastro/astro/pull/4138

## Testing

See https://github.com/withastro/astro/pull/4138 and `test/fixtures/mdx-astro-markdown-remarkRehype/src/pages/index.mdx`

## Docs

/cc @withastro/maintainers-docs for feedback!

### CODE REVIEW

File: packages/integrations/mdx/src/index.ts [hunk 1]
```diff
@@ -71,6 +71,7 @@ export default function mdx(mdxOptions: MdxOptions = {}): AstroIntegration {
 					// Note: disable `.md` (and other alternative extensions for markdown files like `.markdown`) support
 					format: 'mdx',
 					mdExtensions: [],
+					remarkRehypeOptions: config.markdown.remarkRehype,
```
Comment (bholmesdev) [inline packages/integrations/mdx/src/index.ts:74]: I'm noticing this applies `remarkRehype` _in all cases._ To stay consistent with `remark` and `rehype` plugin configs, we should only apply this when `mdxOptions.extendPlugins === 'markdown'`. I'd add a check for that here.

Comment (bholmesdev) [reply packages/integrations/mdx/src/index.ts:74]: I'd probably add `remarkRehype` to the `MdxOptions` as well, in keeping with how we mirror the markdown config for the MDX integration. A couple changes for this:
- Add `remarkRehype` to the `MdxOptions` type
- Spread `mdxOptions.remarkRehype` with `config.markdown.remarkRehype`, checking that `extendPlugins` option I noted above
- Add `remarkRehype` to the MDX README. Feel free to [link to our existing `remarkRehype` docs](https://docs.astro.build/en/reference/configuration-reference/#markdownremarkrehype) for simplicity.

A bit complex I know, and we _will_ simplify these APIs in the future! Just want to make sure we stay consistent with extending Markdown configs into MDX.

**Review by bholmesdev — COMMENTED** [review]
Left a note on adding `remarkRehype` to the MDX config. Great PR though, with tests to boot 🙌

File: packages/integrations/mdx/src/index.ts (hunk 1 above)
Comment (backflip) [reply packages/integrations/mdx/src/index.ts]: @bholmesdev, thanks for the feedback! See latest commits for an attempt to take care of this.

Three things I'm unsure of:
- Is `remarkRehype` of type `RemarkRehype` or `RemarkRehypeOptions`? I would assume the latter, but the `markdown` options assume the former.
- Should `mdxOptions.remarkRehype` be extended with `config.markdown.remarkRehype` or the other way round?
- Is the README helpful enough?

File: packages/integrations/mdx/README.md [hunk 2]
```diff
@@ -518,6 +518,12 @@ These are plugins that modify the output [estree](https://github.com/estree/estr
 
 We suggest [using AST Explorer](https://astexplorer.net/) to play with estree outputs, and trying [`estree-util-visit`](https://unifiedjs.com/explore/package/estree-util-visit/) for searching across JavaScript nodes.
 
+### remarkRehype
+
+Options for `remarkRehype`. See [markdown.remarkRehype](https://docs.astro.build/en/reference/configuration-reference/#markdownremarkrehype).
+
+When `extendPlugins` is set to `'markdown'` (default), this config is extended with `remarkRehype` from Astro's default markdown config.
```
Comment (sarah11918) [inline packages/integrations/mdx/README.md:525]:
```suggestion
### remarkRehype

Markdown content is transformed into HTML through remark-rehype which has [a number of options](https://github.com/remarkjs/remark-rehype#options).

You can use remark-rehype options in your MDX integration config file like so:

```js
// astro.config.mjs
export default {
  integrations: [mdx({
    remarkRehype: {
      footnoteLabel: 'Catatan kaki',
      footnoteBackLabel: 'Kembali ke konten',
    },
  })],
};
```
This inherits the configuration of [`markdown.remarkRehype`](https://docs.astro.build/en/reference/configuration-reference/#markdownremarkrehype). This behavior can be changed by configuring `extendPlugins`.
```
Going by what the Markdown config example looks like, what about sticking with a similar format, which is also in keeping with the remark and rehype plugin options? How does something along these lines look?  (Edit to taste.)

Comment (bholmesdev) [reply packages/integrations/mdx/README.md:525]: Thanks @sarah11918! Change looks good to me 👍

File: packages/integrations/mdx/src/index.ts (hunk 1 above)
Comment (backflip) [reply packages/integrations/mdx/src/index.ts]: @bholmesdev, sounds good!

1. After finding https://github.com/mdx-js/mdx/blob/aff6de4fad5955b804d51cd109248425d786fb1c/packages/mdx/lib/core.js#L7 I have kept `RemarkRehypeOptions`.
2. Fixed and additional tests added.
3. Fixed.  Thanks a lot,  @sarah11918!

File: packages/integrations/mdx/README.md (hunk 2 above)
Comment (backflip) [reply packages/integrations/mdx/README.md]: I have re-added the last line as GitHub dropped it together with the closing ``` for some reason.

**Review by bholmesdev — APPROVED** [review]
These changes look great! Thanks so much 👍 
Also looks like the automated test failures are unrelated. I'll try rerunning, or just merge if we feel confident enough.

**Pull request merged**